cobalt/updater/patcher.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef COBALT_UPDATER_PATCHER_H_
+#define COBALT_UPDATER_PATCHER_H_
+
+#include <memory>
+
+#include "base/memory/scoped_refptr.h"
+
+#include "base/memory/ref_counted.h"
+#include "components/update_client/patcher.h"
+
+// TODO(b/174699165): Investigate differential updates
+
+namespace cobalt {
+namespace updater {
+
+class PatcherFactory : public update_client::PatcherFactory {
```

Comment:
This is a surprising class, because it only creates
PatcherImpls (in the .cc) file), and AFAIK that class
itself is just three methods, no member variables.
In an ideal world, they would be just free functions;
however, the base classes make me think there's more
to it? Could you leave a comment on each class stating
why they are needed? 
It can be short, like:
```
// PatcherFactory is a trivial factory class, needed
// by XYZ to operate.
```
And similarly for PatcherImpl.

Same comments on `UnzipperFactory`